operator/internal/manifests/gateway_test.go

```diff
@@ -321,51 +340,70 @@ func TestBuildGateway_WithTLSProfile(t *testing.T) {
 		},
 		{
 			desc: "dynamic mode",
-			mode: lokiv1.Dynamic,
+			options: Options{
+				Name:      "abcd",
+				Namespace: "efgh",
+				Gates: configv1.FeatureGates{
+					LokiStackGateway: true,
+					HTTPEncryption:   true,
+					TLSProfile:       string(configv1.TLSProfileOldType),
+				},
+				TLSProfileSpec: configv1.TLSProfileSpec{
+					MinTLSVersion: "min-version",
+					Ciphers:       []string{"cipher1", "cipher2"},
+				},
+				Stack: lokiv1.LokiStackSpec{
+					Template: &lokiv1.LokiTemplateSpec{
+						Gateway: &lokiv1.LokiComponentSpec{
+							Replicas: rand.Int31(),
+						},
+					},
+					Tenants: &lokiv1.TenantsSpec{
+						Mode: lokiv1.Dynamic,
+					},
+				},
+			},
 			expectedArgs: []string{
 				"--tls.min-version=min-version",
 				"--tls.cipher-suites=cipher1,cipher2",
 			},
 		},
 		{
 			desc: "openshift-logging mode",
-			mode: lokiv1.OpenshiftLogging,
+			options: Options{
+				Name:      "abcd",
+				Namespace: "efgh",
+				Gates: configv1.FeatureGates{
+					LokiStackGateway: true,
+					HTTPEncryption:   true,
+					TLSProfile:       string(configv1.TLSProfileOldType),
+				},
+				TLSProfileSpec: configv1.TLSProfileSpec{
+					MinTLSVersion: "min-version",
+					Ciphers:       []string{"cipher1", "cipher2"},
+				},
+				Stack: lokiv1.LokiStackSpec{
+					Template: &lokiv1.LokiTemplateSpec{
+						Gateway: &lokiv1.LokiComponentSpec{
+							Replicas: rand.Int31(),
+						},
+					},
+					Tenants: &lokiv1.TenantsSpec{
+						Mode: lokiv1.OpenshiftLogging,
+					},
+				},
+			},
 			expectedArgs: []string{
 				"--tls.min-version=min-version",
 				"--tls.cipher-suites=cipher1,cipher2",
 			},
 		},
 	}
-
-	options := Options{
-		Name:      "abcd",
-		Namespace: "efgh",
-		Gates: configv1.FeatureGates{
-			LokiStackGateway: true,
-			HTTPEncryption:   true,
-			TLSProfile:       string(configv1.TLSProfileOldType),
-		},
-		TLSProfileSpec: configv1.TLSProfileSpec{
-			MinTLSVersion: "min-version",
-			Ciphers:       []string{"cipher1", "cipher2"},
-		},
-		Stack: lokiv1.LokiStackSpec{
-			Template: &lokiv1.LokiTemplateSpec{
-				Gateway: &lokiv1.LokiComponentSpec{
-					Replicas: rand.Int31(),
-				},
-			},
-			Tenants: &lokiv1.TenantsSpec{},
-		},
-	}
-
-	for _, tc := range tc {
+	for _, tc := range tt {
 		tc := tc
 		t.Run(tc.desc, func(t *testing.T) {
 			t.Parallel()
-			options.Stack.Tenants.Mode = tc.mode
-			options.Stack.Tenants.Authorization = tc.authZ
-			objs, err := BuildGateway(options)
+			objs, err := BuildGateway(tc.options)
```

Comment:
Not really related to the change in this PR, but we might also want to change the two `require.Contains` lines below into something that works for any length of `expectedArgs` to avoid confusion in the future.